daemon/cluster/executor/container/controller.go

```diff
@@ -315,12 +345,18 @@ func (r *controller) Shutdown(ctx context.Context) error {
 		r.cancelPull()
 	}
 
-	// remove container from service binding
-	if err := r.adapter.deactivateServiceBinding(); err != nil {
-		log.G(ctx).WithError(err).Warningf("failed to deactivate service binding for container %s", r.adapter.container.name())
-		// Don't return an error here, because failure to deactivate
-		// the service binding is expected if the container was never
-		// started.
+	if r.hasServiceBinding() {
```

Comment:
I like the idea https://github.com/docker/docker/issues/30321#issuecomment-280734751. @sanimej told me it's difficult to support errors from `r.adapter.deactivateServiceBinding` to tell if network convergency is required. That's why this is proposed.